python-threatexchange/threatexchange/signal_type/index.py

```diff
@@ -126,3 +126,8 @@ def serialize(self, fout: t.BinaryIO) -> None:
     def deserialize(cls, fin: t.BinaryIO) -> "SignalTypeIndex[T]":
         """Instanciate an index from a previous call to serialize"""
         raise NotImplementedError
+
+    @property
+    def data_type(cls) -> str:
+        """The Threat Exchange Data type this index type can process"""
+        raise NotImplementedError
```

Comment:
Blocking: What is a data type? Do you mean indicator type?

This is deliberately indirected, since it's possible you might store multiple indicator types in one index if they are serialized differently.

Instead use SignalType -> Index mapping.

Reply:
Hmm. What would be an example of multiple indicator types using the same index? HASH_PDQ and HASH_PDQ_OCR? HASH_PDQ and HASH_TMK?

Reply:
moved to signal type